great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -827,7 +823,12 @@ class SQLDatasource(Datasource):
 
     # private attrs
     _cached_connection_string: str = pydantic.PrivateAttr("")
-    _engine: Union[sqlalchemy.engine.Engine, None] = pydantic.PrivateAttr(None)
+    # The sqlalchemy is in quotes (hence the # noqa) here because otherwise pydantic
+    # will try to resolve this annotation at gx import time which will cause the
+    # process die if sqlalchemy is not installed.
+    _engine: Union[
+        "sqlalchemy.engine.Engine", None  # noqa: UP037
+    ] = pydantic.PrivateAttr(None)
```

Comment:
I would expect this to work as long as you are doing a `IF TYPE_CHECKING: import sqlalchemy` and using `from __future__ import annotations`

Reply:
So is this going to force us to always use strings anytime we use this `NotImported`? That seems less than ideal.

The problem with string type annotations is that it's very easy for them to turn into `Any`'s.
If `sqlalchemy` isn't available when the type-checker is run then this type will turn into an `Any` and it will be hard to catch when and if this has happened and easy for typing bugs to get through.
Example: `sqlalchemy` isn't installed as part of the type-checking step, or if this import gets removed.

When we use non-string post-ponned annotations it's easy to catch, ruff will flag that symbol/name is not available.

Discussed this same point regarding some ruff auto-fixes a little while ago.
https://github.com/charliermarsh/ruff/issues/2329#issuecomment-1415930582